cosmo_tester/test_suites/manager_of_managers/tier_1_clusters.py

```diff
@@ -0,0 +1,455 @@
+            local_cert,
+            use_sudo=True
+        )
+        return local_cert
+
+    def validate(self):
+        """
+        For Floating IP clusters validation involves creating a REST client
+        to connect to the master manager, and making sure that certain
+        Cloudify resources (tenants, plugins, etc) were created
+        """
+        self._validate_tenants_created()
+        self._validate_blueprints_created()
+        self._validate_deployments_created()
+        self._validate_secrets_created()
+        self._validate_plugins_created()
+
+    def _validate_tenants_created(self):
+        self.logger.info(
+            'Validating that tenants were created on Tier 1 cluster...'
+        )
+        tenants = self.client.tenants.list(_include=['name'])
+        tenant_names = {t['name'] for t in tenants}
+        assert tenant_names == {DEFAULT_TENANT_NAME,
+                                constants.TENANT_1,
+                                constants.TENANT_2}
+        self.logger.info('Tenants validated successfully')
+
+    def _validate_blueprints_created(self):
+        self.logger.info(
+            'Validating that blueprints were created on Tier 1 cluster...'
+        )
+        blueprints = self.client.blueprints.list(
+            _all_tenants=True,
+            _include=['id', 'tenant_name']
+        )
+        blueprint_pairs = {(b['id'], b['tenant_name']) for b in blueprints}
+        assert blueprint_pairs == {
+            (
+                'cloudify-hello-world-example-4.5.no-monitoring-singlehost-blueprint',  # NOQA
+                DEFAULT_TENANT_NAME
+            ),
+            ('second_bp', constants.TENANT_2),
+            (constants.HELLO_WORLD_BP, constants.TENANT_1)
+        }
+        self.logger.info('Blueprints validated successfully')
+
+    def _validate_deployments_created(self):
+        self.logger.info(
+            'Validating that deployments were created on Tier 1 cluster...'
+        )
+        deployments = self.client.deployments.list(
+            _all_tenants=True,
+            _include=['id', 'blueprint_id']
+        )
+        assert len(deployments) == 1
+        deployment = deployments[0]
+        assert deployment.id == constants.HELLO_WORLD_DEP
+        assert deployment.blueprint_id == constants.HELLO_WORLD_BP
+
+        self.logger.info('Deployments validated successfully')
+
+    def _validate_secrets_created(self):
+        self.logger.info(
+            'Validating that secrets were created on Tier 1 cluster...'
+        )
+        secrets = self.client.secrets.list(_all_tenants=True)
+        secrets = {s['key']: s for s in secrets}
+
+        expected_set = {constants.SECRET_FILE_KEY, constants.SECRET_STRING_KEY}
+
+        # During upgrade we add secrets for ssh keys, so the actual set might
+        # not be equal exactly, but may contain extra values
+        assert set(secrets.keys()).issuperset(expected_set)
```

Comment:
maybe print the diff as info for future debugging purposes

Reply:
I'm pretty sure AssertionErrors include this info